Singleton-ThreadPool/src/ThreadPool.java

```diff
@@ -0,0 +1,29 @@
+public class ThreadPool {
+
+    private static int size = 10;
+    private static volatile ThreadPool threadPoolObject = null;
```

Comment:
what is the necessity of making this volatile? threadPoolObject modification is synchronized?

Reply:
We have declared the obj vola which ensures that multiple threads offer the obj variable correctly when it is being initialized to Singleton instance.

I have synchronized threadPoolObject If I am not wrong